### ISSUE TITLE
errno 1 error - please help

### ISSUE THREAD

**Issue by MyCryptoBabies** [HashLips/hashlips_art_engine#915]
Everyday it is something new.  I don't change anything so I don't understand how It can work and generate.  I go to bed, wake up and it doesn't work....  anyway,  I got my things generated (at least a batch to get started) changing the base uri I get this after running 
npm run update_info.  
What?  Daniel makes it looks so smooth on the videos.  I am on a windows 10
![+errno 1](https://user-images.githubusercontent.com/100310137/157327924-8d0f4fdd-efbc-48da-ad97-17c65b668927.PNG)

**Comment by bolshoytoster** [comment]
@MyCryptoBabies can you paste the full error please.

**Comment by MyCryptoBabies** [comment]
Man your quick - I like it when you are working...  
![errno 1 1](https://user-images.githubusercontent.com/100310137/157329520-a6e6f64e-e201-4056-976d-cb3df0ce94b0.PNG)

![errno 1 2](https://user-images.githubusercontent.com/100310137/157329530-6d1201f4-5051-46b6-be8b-b1a1770ad9fa.PNG)

![errno 1 3](https://user-images.githubusercontent.com/100310137/157329540-66bbb167-9ed7-4cd7-ad28-09f5df1b921b.PNG)

![errno 1 4](https://user-images.githubusercontent.com/100310137/157329550-98dc65b1-7c4f-4f53-aac9-2579baab6a82.PNG)

**Comment by MyCryptoBabies** [comment]
It looks like I didn't put the metadata in there.  I am following Daniels new video step by step

**Comment by bolshoytoster** [comment]
@MyCryptoBabies In this repo, it doesn't create the `_metadata.json` file if it runs into too many duplicates (if it says `You need more layers or elements to grow your edition to ... artworks!`). You could either use [my repo](https://github.com/bolshoytoster/hashlips_art_engine), or you could add this line:
```js
writeMetaData(JSON.stringify(metadataList, null, 2));
```
At ~line 423 of `src/main.js`:
https://github.com/HashLips/hashlips_art_engine/blob/d8ee279043d2d4a8de3bdfac0d89d0e966fb04a2/src/main.js#L422-L423

Insert it just before the `process.exit();`

Either way, you'll have to regenerate your collection.

**Comment by MyCryptoBabies** [comment]
I changed some of the pictures numbers and in the json file to match.  That must have messed it up. I went through all the pictures and didn't like some so I took them out.

**Comment by bolshoytoster** [comment]
@MyCryptoBabies I don't think that would cause it; unless you deleted the `build/json/_metadata.json` file.

**Comment by MyCryptoBabies** [comment]
trying to generate again from scratch - getting this . I am just so frustrated because I have been working on all this for 6 weeks now and all I want to do is get going.  Thanks for your help.

![+errno 1](https://user-images.githubusercontent.com/100310137/157335793-1432695e-884c-4b0d-9104-1ea4b329322d.PNG)

**Comment by MyCryptoBabies** [comment]
trying to generate again from scratch - getting this . I am just so frustrated because I have been working on all this for 6 weeks now and all I want to do is get going.  Thanks for your help.

![+errno 1](https://user-images.githubusercontent.com/100310137/157335793-1432695e-884c-4b0d-9104-1ea4b329322d.PNG)

**Comment by bolshoytoster** [comment]
@MyCryptoBabies I think you're in the wrong folder, run `Get-ChildItem -Directory` to get the name, then run `cd ` + name to go into it.

**Comment by MyCryptoBabies** [comment]
I went into the json pkg and pushed the debug button.  It's generating now.  What happened?

**Comment by bolshoytoster** [comment]
@MyCryptoBabies I have no clue. By 'json pkg' do you mean `package.json`?

**Comment by MyCryptoBabies** [comment]
got 336 and it stopped then got this

![+errno 1](https://user-images.githubusercontent.com/100310137/157337868-ddad056a-fa99-4987-a0a6-28bebdeae63b.PNG)

**Comment by MyCryptoBabies** [comment]
package.json  yes, that... I hit debug again and it is generating more

**Comment by bolshoytoster** [comment]
@MyCryptoBabies if you run `node index.js` instead of `npm run generate` it'll probably show the error.

**Comment by MyCryptoBabies** [comment]
i did that and it didn't show an error.  It started generating.  BTW - Can i rename these files if I rename it in the json file that matches?  If so, can I do it in windows explorer or do I have to do it in the visual studio.?

**Comment by bolshoytoster** [comment]
@MyCryptoBabies if you rename the files, you can rename the pngs, then go to `build/json/_metadata.json` and change the corresponding `edition` value.

> If so, can I do it in windows explorer or do I have to do it in the visual studio.?

It doesn't matter.

**Comment by MyCryptoBabies** [comment]
I don't see a metadata file in the build json folder on these runs

**Comment by bolshoytoster** [comment]
@MyCryptoBabies did your build end with `You need more layers or elements to grow your edition to ... artworks!`?

**Comment by MyCryptoBabies** [comment]
no,  I have a lot of layers and images - 7 layers and at least 20-50 in each folder

**Comment by MyCryptoBabies** [comment]
no,  I have a lot of layers and images - 7 layers and at least 20-50 in each folder

**Comment by MyCryptoBabies** [comment]
I just don't want to waste MORE time renaming if I need the metadata json file.  I'm not sure why it's not there

**Comment by MyCryptoBabies** [comment]
I just don't want to waste MORE time renaming if I need the metadata json file.  I'm not sure why it's not there

**Comment by bolshoytoster** [comment]
@MyCryptoBabies paste this code:
```js
const basePath = process.cwd();
const { readdirSync, readFileSync, writeFileSync } = require("fs");
const buildDir = `${basePath}/build`;

writeFileSync(
  `${basePath}/build/json/_metadata.json`,
  JSON.stringify(
    readdirSync(`${buildDir}/json`)
      .map((name) => JSON.parse(readFileSync(`${buildDir}/json/${name}`))),
    null, 2
  )
);
```
into a file in your project folder, then run it with `node ` + whatever you called the file.

The code will generate a `_metadata.json`.

**Comment by MyCryptoBabies** [comment]
Why would it just stop working if no changes are made?  I generated some.  Tried to generate more and it gives me this.  The only thing I did is move the last build into another folder on my desktop to start from scratch with a new collection - I'm in the config.js folder

![+errno 1](https://user-images.githubusercontent.com/100310137/157344311-6e61e3b5-1244-4ce2-87fb-fac9bec52b03.PNG)

**Comment by MyCryptoBabies** [comment]
Why would it just stop working if no changes are made?  I generated some.  Tried to generate more and it gives me this.  The only thing I did is move the last build into another folder on my desktop to start from scratch with a new collection - I'm in the config.js folder

![+errno 1](https://user-images.githubusercontent.com/100310137/157344311-6e61e3b5-1244-4ce2-87fb-fac9bec52b03.PNG)

**Comment by bolshoytoster** [comment]
@MyCryptoBabies if you're using my fork, it normally carries on from the previous build. So if you want to restart you have to manually delete the `build` folder.

**Comment by MyCryptoBabies** [comment]
Where and how do I get your fork? And how to I use it?  The Problem is nothing will generate now.  Should I delete the program and reinstall it?
